sdgym/summary.py

```diff
@@ -190,12 +201,41 @@ def add_sheet(dfs, name, writer, cell_fmt, index_fmt, header_fmt):
         worksheet.set_column(idx, idx, width + 1, fmt)
 
 
+def _find_category(synthesizer, categories):
+    for category, category_synthesizers in categories.items():
+        for category_synthesizer in category_synthesizers:
+            if category_synthesizer in synthesizer.lower():
+                return category
+
+    return None
+
+
+def _add_summary_categories(summary_data):
+    categories = {
+        'SDV': ['ctgan', 'copulagan', 'gaussiancopula', 'tvae', 'hma1', 'par'],
+        'Gretel': ['gretel'],
+        'YData': ['dragan', 'vanilllagan', 'wgan'],
+    }
+    summary_data['category'] = summary_data.index.map(
+        lambda x: _find_category(x, categories))
```

Comment:
The `lambda` is not needed here. If we still need to pass down categories anyway (see coment above), we can use `apply` instead of `map` and pass the `categories` kwarg down directly:

```python3
summary_data['category'] = summary_data.index.apply(_find_category, categories=categories)
```